src/GuildsApp.tsx

```diff
@@ -31,30 +39,46 @@ const GuildsApp = () => {
       <HashRouter basename="/guilds">
         <GlobalErrorBoundary>
           <WalletWeb3Manager>
-            <GlobalStyle />
-            <Header />
-            <Container>
-              <Switch>
-                <Redirect
-                  exact
-                  from="/"
-                  to="/rinkeby/0x9cdc16b5f95229b856cba5f38095fd8e00f8edef"
-                />
-                <Redirect
-                  exact
-                  from="/:chain_name"
-                  to="/:chain_name/0x9cdc16b5f95229b856cba5f38095fd8e00f8edef"
-                />
-                <Route exact path="/:chain_name/:guild_id">
-                  <GuildsContextProvider>
-                    <GuildsPage />
-                  </GuildsContextProvider>
-                </Route>
-                <Route path="/:chain_name/:guild_id/proposals/:proposal_id">
-                  <ProposalPage />
-                </Route>
-              </Switch>
-            </Container>
+            <EthSWRConfig
+              value={{
+                web3Provider: provider,
+                ABIs: new Map([
+                  [
+                    // If we need to generate this for each guildId we may need this wrapping after each route
+                    '0x9cdc16b5f95229b856cba5f38095fd8e00f8edef',

```

Comment:
So we are definitely going to need this to be dynamic. We are also going to need the abi to be potentially different in future.
We will have some different variances and implementations ontop of the base erc20 contract.

Reply:
So when we useSWR we define which ABI is used by the guild Id right?